qemu/tests/nested_test.py

```diff
@@ -0,0 +1,125 @@
+import os
+import logging
+import tempfile
+import time
+import json
+
+from avocado.utils import process
+from avocado.utils import software_manager
+
+from virttest import error_context
+from virttest import data_dir as virttest_data_dir
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Nested test:
+    1) Boot VM
+    2) Install ansible and related packages
+    2) Generate inventory file with L1 guest IP
+    3) Generate json file to save test parameters
+    4) Execute ansible command
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    setup_bridge_sh = "/root/setup_bridge.sh"
+
+    def get_live_vms(env):
+        live_vms = []
+        for vm in env.get_all_vms():
+            if vm.is_alive():
+                live_vms.append(vm)
+        if not live_vms:
+            test.fail("No live VM.")
+        return live_vms
+
+    def generate_invent_file(env):
+        vms = get_live_vms(env)
+        tmp_dir = virttest_data_dir.get_tmp_dir()
+        invent_file = tempfile.mktemp(dir=tmp_dir)
```

Comment:
`tempfile.mktemp` is deprecated since python2.3, please use `tempfile.mkstemp`, or you can just create a regular file instead.
https://docs.python.org/3.6/library/tempfile.html?highlight=mktemp#tempfile.mktemp

Reply:
Create regular file.